plugins/woocommerce-blocks/assets/js/blocks/product-collection/utils.tsx

```diff
@@ -64,6 +64,15 @@ const isInProductArchive = () => {
 		'woocommerce/woocommerce//taxonomy-product_tag',
 		'woocommerce/woocommerce//taxonomy-product_attribute',
 		'woocommerce/woocommerce//product-search-results',
+		// Custom taxonomy templates have structure:
+		// <<THEME>>//taxonomy-product_cat-<<CATEGORY>>
+		// hence we're checking if template ID includes the middle part.
+		//
+		// That includes:
+		// - woocommerce/woocommerce//taxonomy-product_cat
+		// - woocommerce/woocommerce//taxonomy-product_tag
+		'//taxonomy-product_cat',
+		'//taxonomy-product_tag',
```

Comment:
I believe it would be better to add `-` in the end of the template paths:
```
//taxonomy-product_cat-
//taxonomy-product_tag-
```

This way, it's more specific and less likely to have false positives.

In case you think we shouldn't add `-`, we can consider removing `woocommerce/woocommerce//taxonomy-product_cat` and `woocommerce/woocommerce//taxonomy-product_tag` from the list. Because `//taxonomy-product_cat` and `//taxonomy-product_tag` will match them as well.

Reply:
> Because //taxonomy-product_cat and //taxonomy-product_tag will match them as well.

Ah, that was my intention 😅 Thanks for catching it!